backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlDatabase.java

```diff
@@ -86,7 +88,8 @@ private void setJooq(DSLContext ctx) {
   public SqlDatabase(boolean init) {
     initDataSource();
     this.connectionProvider = new SqlUserAwareConnectionProvider(source);
-    this.jooq = DSL.using(connectionProvider, SQLDialect.POSTGRES);
+    final Settings settings = new Settings().withQueryTimeout(TEN_SECONDS);
```

Comment:
this should only apply to queries, not to transactions because many data uploads take longer. So I suppose we should try to inject this when firing a query? Or does this only apply to queries?

Reply:
I have been reading about it, we can also set this only in SqlQuery classes

Reply:
why do you think this would apply to transactions ?

Reply:
Maybe it doesn't. I assume this timeout only applies to individual statements. I suppose the 'commit' is not such a statement. All other statements (including our inserts/updates) should take much shorter, generally under 1 second; large transaction that we run partition the statements.